flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -904,6 +906,8 @@ public void heartbeatFromResourceManager(ResourceID resourceID) {
 			case STDOUT:
 				filePath = taskManagerConfiguration.getTaskManagerStdoutPath();
 				break;
+			case THREAD_DUMP:
+				return putTransientBlobStream(JvmUtils.threadDumpStream(), fileType);
```

Comment:
I think it would be good to add a test for the thread dump feature (e.g. testing that `TaskExecutorGateway#requestThreadDump()` will return the blob key to a thread dump file.